GitHubActionsTestLogger/CliOptionsProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Testing.Platform.CommandLine;
+using Microsoft.Testing.Platform.Extensions;
+using Microsoft.Testing.Platform.Extensions.CommandLine;
+
+namespace GitHubActionsTestLogger;
+
+internal sealed class CliOptionsProvider(GitHubTestReporterExtension extension) : ICommandLineOptionsProvider
+{
+    public const string ReportGitHubOption = "report-github";
+    public const string ReportGitHubSummaryOption = "report-github-summary";
+    public const string ReportGitHubTitleOption = "report-github-title";
+    public const string ReportGitHubMessageOption = "report-github-message";
+
+    public static class ReportGitHubSummaryArguments
+    {
+        public const string IncludePassedTests = "includePassedTests";
+        public const string IncludeSkippedTests = "includeSkippedTests";
+        public const string IncludeNotFoundTests = "includeNotFoundTests";
+    }
+
+    public string Uid => extension.Uid;
+    public string Version => extension.Version;
+    public string DisplayName => extension.DisplayName;
+    public string Description => extension.Description;
+
+    public IReadOnlyCollection<CommandLineOption> GetCommandLineOptions() =>
+        [
+            new(ReportGitHubOption, "Reports test run information to GitHub Actions", ArgumentArity.Zero, isHidden: false),
+            // TODO: Do you prefer multiple zero argument options or a single option with argument?
+            new(ReportGitHubSummaryOption, "Defines the information to include in the summary", ArgumentArity.OneOrMore, isHidden: false),
+            new(ReportGitHubTitleOption, "Defines the annotation title format used for reporting test failures", ArgumentArity.ExactlyOne, isHidden: false),
+            new(ReportGitHubMessageOption, "Defines the annotation message format used for reporting test failures", ArgumentArity.ExactlyOne, isHidden: false),
+        ];
+
+    public Task<bool> IsEnabledAsync() => extension.IsEnabledAsync();
+
+    // This method is called once after all options are parsed and is used to validate the combination of options.
+    public Task<ValidationResult> ValidateCommandLineOptionsAsync(ICommandLineOptions commandLineOptions)
+    {
+        // We just want to validate that the sub options are set only if the main option is set.
+        if (commandLineOptions.IsOptionSet(ReportGitHubOption))
+        {
+            return ValidationResult.ValidTask;
+        }
+
+        if (commandLineOptions.IsOptionSet(ReportGitHubSummaryOption)
+            || commandLineOptions.IsOptionSet(ReportGitHubTitleOption)
+            || commandLineOptions.IsOptionSet(ReportGitHubMessageOption))
```

Comment:
Would it be possible to refer to the list of available options via `GetCommandLineOptions()` here so as to avoid having multiple sources of truths?